curiosrc/web/api/sector/sector.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/filecoin-project/lotus/curiosrc/web/api/apihelper"
+	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+)
+
+type cfg struct {
+	*deps.Deps
+}
+
+func Routes(r *mux.Router, deps *deps.Deps) {
+	c := &cfg{deps}
+	// At menu.html:
+	r.Methods("GET").Path("/all").HandlerFunc(c.getSectors)
+	r.Methods("POST").Path("/terminate").HandlerFunc(c.terminateSectors)
+}
+
+func (c *cfg) terminateSectors(w http.ResponseWriter, r *http.Request) {
+	var in []struct {
+		MinerID int
+		Sector  int
+	}
+	apihelper.OrHTTPFail(w, json.NewDecoder(r.Body).Decode(&in))
+	toDel := map[int][]int{}
+	for _, s := range in {
+		toDel[s.MinerID] = append(toDel[s.MinerID], s.Sector)
+	}
+
+	for minerInt, sectors := range toDel {
+		maddr, err := address.NewIDAddress(uint64(minerInt))
+		apihelper.OrHTTPFail(w, err)
+		mi, err := c.Full.StateMinerInfo(r.Context(), maddr, types.EmptyTSK)
+		apihelper.OrHTTPFail(w, err)
+		_, err = spcli.TerminateSectors(r.Context(), c.Full, maddr, sectors, mi.Worker)
+		apihelper.OrHTTPFail(w, err)
+		for _, sectorNumber := range sectors {
+			id := abi.SectorID{Miner: abi.ActorID(minerInt), Number: abi.SectorNumber(sectorNumber)}
+			apihelper.OrHTTPFail(w, c.Stor.Remove(r.Context(), id, storiface.FTAll, true, nil))
+		}
+	}
+}
+
+func (c *cfg) getSectors(w http.ResponseWriter, r *http.Request) {
+	// TODO get sector info from chain and from database, then fold them together
+	// and return the result.
+	type sector struct {
+		MinerID        int64 `db:"miner_id"`
+		SectorNum      int64 `db:"sector_num"`
+		SectorFiletype int   `db:"sector_filetype" json:"-"` // Useless?
+		HasSealed      bool
+		HasUnsealed    bool
+		HasSnap        bool
+		ExpiresAt      abi.ChainEpoch // map to Duration
+		IsOnChain      bool
+		//StorageID         string         `db:"storage_id"` // map to serverName
+		// Activation        abi.ChainEpoch // map to time.Time. advanced view only
+		// DealIDs           []abi.DealID //  advanced view only
+		ExpectedDayReward abi.TokenAmount
+		IsFilPlus         bool
+		SealProof         abi.RegisteredSealProof
+		SealInfo          string
+	}
+	var sectors []sector
+	apihelper.OrHTTPFail(w, c.DB.Select(r.Context(), &sectors, `SELECT 
+		miner_id, sector_num, SUM(sector_filetype) as sector_filetype  
+		FROM sector_location 
+		GROUP BY miner_id, sector_num 
+		ORDER BY miner_id, sector_num`))
+	minerToAddr := map[int64]address.Address{}
+	head, err := c.Full.ChainHead(r.Context())
+	apihelper.OrHTTPFail(w, err)
+
+	type sectorID struct {
+		mID  int64
+		sNum uint64
+	}
+	sectorIdx := map[sectorID]int{}
+	for i, s := range sectors {
+		sectors[i].HasSealed = s.SectorFiletype&int(storiface.FTSealed) != 0 || s.SectorFiletype&int(storiface.FTUpdate) != 0
+		sectors[i].HasUnsealed = s.SectorFiletype&int(storiface.FTUnsealed) != 0
+		sectors[i].HasSnap = s.SectorFiletype&int(storiface.FTUpdate) != 0
+		if ss, err := s.SealProof.SectorSize(); err == nil {
+			sectors[i].SealInfo = ss.ShortString()
+		}
```

Comment:
This should be moved further down to when we have chainInfo. Currently, this displays wrong value.